backend/app/Repositories/EloquentChartVisitorRepository.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repositories;
+
+use App\Contracts\Visitors\VisitorsBounceRateFilterData;
+use App\Entities\Visitor;
+use App\Model\Visitors\VisitorsBounceRateResponseItem;
+use App\Repositories\Contracts\ChartVisitorRepository;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Collection;
+
+final class EloquentChartVisitorRepository implements ChartVisitorRepository
+{
+    public function getBounceRateCollection(VisitorsBounceRateFilterData $filterData): Collection
+    {
+        $from = $filterData->getStartDate();
+        $to = $filterData->getEndDate();
+        $timeFrame = $filterData->getTimeFrame();
+        $allVisitorsByTimeFrame = Visitor::whereCreatedAtBetween($from, $to)
+            ->selectRaw('COUNT (*)')
+            ->selectRaw(' (extract(epoch FROM created_at) - MOD( (CAST (extract(epoch FROM created_at) AS INTEGER)), ? )) AS period', [$timeFrame])
+            ->groupBy('period')
+            ->get();
+
+
+        $allVisitorsByTimeFrameValues = array_column($allVisitorsByTimeFrame->toArray(), 'count', 'period');
+
+        $bounceVisitorsByTimeFrame =  Visitor::whereCreatedAtBetween($from, $to)
+            ->has('sessions', '=', '1')
+            ->whereHas('sessions', function (Builder $query) use ($from, $to) {
+                $query->whereBetween('start_session', [$from, $to])
```

Comment:
I don't think that visitors created in date range can have sessions that started before they were created)